framework/src/outputs/TopResidualDebugOutput.C

```diff
@@ -55,23 +55,24 @@ TopResidualDebugOutput::output(const ExecFlagType & /*type*/)
 void
 TopResidualDebugOutput::printTopResiduals(const NumericVector<Number> & residual, unsigned int n)
 {
-  // Need a reference to the libMesh mesh object
-  MeshBase & mesh = _problem_ptr->mesh().getMesh();
+  MooseMesh & mesh = _problem_ptr->mesh();
 
   std::vector<TopResidualDebugOutputTopResidualData> vec;
   vec.resize(residual.local_size());
 
   unsigned int j = 0;
-  for (const auto & node : mesh.local_node_ptr_range())
+  for (const auto & node : as_range(mesh.localNodesBegin(), mesh.localNodesEnd()))
   {
     dof_id_type nd = node->id();
 
     for (unsigned int var = 0; var < node->n_vars(_sys.number()); ++var)
       if (node->n_dofs(_sys.number(), var) >
           0) // this check filters scalar variables (which are clearly not a dof on every node)
       {
+        const auto & subdomain_ids = mesh.getNodeBlockIds(*node);
         dof_id_type dof_idx = node->dof_number(_sys.number(), var, 0);
-        vec[j] = TopResidualDebugOutputTopResidualData(var, nd, residual(dof_idx));
+        vec[j] = TopResidualDebugOutputTopResidualData(
+            var, subdomain_ids, nd, Point(*node), residual(dof_idx));
```

Comment:
Once you switch to const ref in the declaration, you won't have to create a temporary point here. You can just pass in the *node directly.